rsmtool/test_utils.py

```diff
@@ -700,3 +706,242 @@ def check_subgroup_outputs(output_dir, experiment_id, subgroups, file_format='cs
+            stem.endswith('_pred_processed') or \
+            stem.endswith('_score_dist')
+
+    def update_source(self, source, skll=False):
+        """
+        Update the test outputs for experiment test with `source` as the
+        given name. It deletes files that are only in the tests directory,
+        adds files that are only in the updated test outputs directory, and
+        updates the files that have changed in the updated test outputs directory.
+        It does not return anything but updates the `deleted_files`, `updated_files`,
+        and `missing_or_empty_sources`  class attributes appropriately.
+
+        Args:
+            source (str): Name of source directory.
+            skll (bool, optional): Whether the given source
+                is for a SKLL-based experiment test.
+
+        """
+        # locate the updated outputs for the experiment under the given
+        # outputs directory and also locate the existing experiment outputs
+        updated_output_path = self.updated_outputs_directory / source / "output"
+        existing_output_path = self.tests_directory / "data" / "experiments" / source / "output"
+
+        # if the directory for this source does not exist on the updated output
+        # side, then that's a problem and something we should report on later
+        try:
+            assert updated_output_path.exists()
+        except AssertionError:
+            self.missing_or_empty_sources.append(source)
+            return
+
+        # if the updated output path does not exist, then create it
+        try:
+            assert existing_output_path.exists()
+        except AssertionError:
+            sys.stderr.write("\nNo existing output for \"{}\". "
+                             "Creating directory ...\n".format(source))
+            existing_output_path.mkdir(parents=True)
+
+        # get a comparison betwen the two directories
+        dir_comparison = dircmp(updated_output_path, existing_output_path)
+
+        # if no output was found in the updated outputs directory, that's
+        # likely to be a problem so save that source
+        if not dir_comparison.left_list:
+            self.missing_or_empty_sources.append(source)
+            return
+
+        # first delete the files that only exist in the existing output directory
+        # since those are likely old files from old versions that we do not need
+        existing_output_only_files = dir_comparison.right_only
+        for file in existing_output_only_files:
+            remove(existing_output_path / file)
+
+        # Next find all the NEW files in the updated outputs but do not include
+        # the config JSON files or the OLS summary files or
+        # the evaluation/prediction/model files for SKLL models
+        new_files = dir_comparison.left_only
+        new_files = [f for f in new_files if not f.endswith('_rsmtool.json') and not f.endswith('_rsmeval.json') and not f.endswith('_ols_summary.txt')]
```

Comment:
Again, would it be better to just use `any()` here and a list of filename suffixes, in case we expand on this in the future?